app/helpers/hosts_and_hostgroups_helper.rb

```diff
@@ -46,28 +46,30 @@ def os_ptable
     @operatingsystem.ptables
   end
 
-  def puppet_master_fields(f)
-    "#{puppet_ca(f)} #{puppet_master(f)}".html_safe
+  def puppet_master_fields(f, can_override = false)
+    "#{puppet_ca(f, can_override)} #{puppet_master(f, can_override)}".html_safe
   end
 
-  def puppet_ca(f)
+  INHERIT_TEXT = N_("inherit")
+
+  def puppet_ca(f, can_override)
     # Don't show this if we have no CA proxies, otherwise always include blank
     # so the user can choose not to sign the puppet cert on this host
     proxies = SmartProxy.unscoped.with_features("Puppet CA").with_taxonomy_scope(@location,@organization,:path_ids)
     return if proxies.count == 0
     select_f f, :puppet_ca_proxy_id, proxies, :id, :name,
-             { :include_blank => blank_or_inherit_f(f, :puppet_ca_proxy) },
+             { :include_blank => blank_or_inherit_f(f, :puppet_ca_proxy), :disable_button => can_override ? INHERIT_TEXT : nil },
```

Comment:
No, it's extracted only.  `N_()` will return the original English string, you then need to pass it back through `_()` here to translate it based on the current locale.